helix-term/src/application.rs

```diff
@@ -1005,20 +1009,69 @@ impl Application {
                                 .err()
                                 .map(|err| err.failed_change_idx as u32),
-                        }))
+                        })))
                     }
                     Ok(MethodCall::WorkspaceFolders) => {
                         let language_server =
                             self.editor.language_servers.get_by_id(server_id).unwrap();
 
-                        Ok(json!(language_server.workspace_folders()))
+                        Some(Ok(json!(language_server.workspace_folders())))
+                    }
+                    Ok(MethodCall::ShowMessageRequest(params)) => {
+                        if let Some(actions) = params.actions {
+                            let call_id = id.clone();
+                            let message_type = match params.typ {
+                                MessageType::ERROR => "ERROR",
+                                MessageType::WARNING => "WARNING",
+                                MessageType::INFO => "INFO",
+                                _ => "LOG",
+                            };
+                            let message = format!("{}: {}", message_type, &params.message);
+                            let rope = Rope::from(message);
+                            let picker = FilePicker::new(
+                                actions,
+                                (),
+                                move |ctx, message_action, _event| {
+                                    let server_from_id =
+                                        ctx.editor.language_servers.get_by_id(server_id);
+                                    let language_server = match server_from_id {
+                                        Some(language_server) => language_server,
+                                        None => {
+                                            warn!(
+                                                "can't find language server with id `{server_id}`"
+                                            );
+                                            return;
+                                        }
+                                    };
+                                    let response = match message_action {
+                                        Some(item) => json!(item),
+                                        None => serde_json::Value::Null,
+                                    };
+
+                                    tokio::spawn(
+                                        language_server.reply(call_id.clone(), Ok(response)),
+                                    );
+                                },
+                                move |_editor, _value| {
+                                    let file_location: FileLocation = (rope.clone().into(), None);
+                                    Some(file_location)
+                                },
+                            );
+                            let popup_id = "show_message_request";
```

Comment:
```suggestion
                            let popup_id = "show-message-request";
```

these are always kebab-case (signature-help, invalid-regex, code-action, ..)